# HG changeset patch
# User Kai T. Ohlhus <k.ohlhus@gmail.com>
# Date 1467809792 -7200
#      Wed Jul 06 14:56:32 2016 +0200
# Node ID d3c79eb1b483cd7e587c963f63d0e5d68822ff44
# Parent  571b42508e1e17def4cb615426c8b09ea0faf256
Add regression test (bug #46330).

* test/module.mk: Add new test file.

* test/bug-46330.tst: Add test case from bug report.

diff --git a/test/bug-46330.tst b/test/bug-46330.tst
new file mode 100644
--- /dev/null
+++ b/test/bug-46330.tst
@@ -0,0 +1,20 @@
+## bug #46330: segfault with matrices containing Inf & NaN
+
+%!warning
+%! a = [-0.46E-01,            0.10681415316, 0.0,   -0.17121680433;
+%!      -0.1675901504661613, -0.515,         1.0,    0.6420630320636088E-02;
+%!       0.1543104215347786, -0.547945,     -0.906, -0.1521689385990753E-02;
+%!       0.0,                 0.0,           1.0,    0.0];
+%!
+%! b = [0.1602300107479095,      0.2111848453E-02;
+%!      0.8196877780963616E-02, -0.3025E-01;
+%!      0.9173594317692437E-01, -0.75283075;
+%!      0.0,                     0.0];
+%!
+%! c = [1.0, 0.0, 0.0, 0.0;
+%!      0.0, 0.0, 0.0, 1.0];
+%!
+%! ## This statement caused an error in LAPACK and eventually caused
+%! ## a segmentation fault.
+%! ## Triggers "warning: matrix singular to machine precision"
+%! assert (c / (i * Inf * eye (4) - a) * b, zeros (2, 2))
diff --git a/test/module.mk b/test/module.mk
--- a/test/module.mk
+++ b/test/module.mk
@@ -5,16 +5,17 @@ test_DISTCLEANFILES =
 test_MAINTAINERCLEANFILES =
 
 TEST_FILES += \
   test/fntests.m \
   test/args.tst \
   test/bug-31371.tst \
   test/bug-38565.tst \
   test/bug-38576.tst \
+  test/bug-46330.tst \
   test/colormaps.tst \
   test/command.tst \
   test/complex.tst \
   test/diag-perm.tst \
   test/error.tst \
   test/eval-catch.tst \
   test/for.tst \
   test/func.tst \
